lib/remap-functions/src/parse_key_value.rs

```diff
@@ -0,0 +1,370 @@
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "key_value_delimiter",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+            Parameter {
+                keyword: "field_delimiter",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+        let key_value_delimiter = arguments.optional("key_value_delimiter").map(Expr::boxed);
+        let field_delimiter = arguments.optional("field_delimiter").map(Expr::boxed);
+
+        Ok(Box::new(ParseKeyValueFn {
+            value,
+            key_value_delimiter,
+            field_delimiter,
+        }))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct ParseKeyValueFn {
+    value: Box<dyn Expression>,
+    key_value_delimiter: Option<Box<dyn Expression>>,
+    field_delimiter: Option<Box<dyn Expression>>,
+}
+
+impl Expression for ParseKeyValueFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let bytes = self.value.execute(state, object)?.try_bytes()?;
+        let value = String::from_utf8_lossy(&bytes);
+
+        let key_value_delimiter = match &self.key_value_delimiter {
+            Some(s) => String::from_utf8_lossy(&s.execute(state, object)?.try_bytes()?).to_string(),
+            None => "=".to_string(),
+        };
+
+        let field_delimiter = match &self.field_delimiter {
+            Some(s) => String::from_utf8_lossy(&s.execute(state, object)?.try_bytes()?).to_string(),
+            None => " ".to_string(),
+        };
+
+        let (_, values) =
+            parse_line(&value, &key_value_delimiter, &field_delimiter).map_err(|e| e.to_string())?;
+
+        Ok(values.into_iter().collect::<BTreeMap<_, _>>().into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .merge_optional(
+                self.key_value_delimiter
+                    .as_ref()
+                    .map(|key_value_delimiter| key_value_delimiter.type_def(state)),
+            )
+            .merge_optional(
+                self.field_delimiter
+                    .as_ref()
+                    .map(|field_delimiter| field_delimiter.type_def(state)),
+            )
+            .into_fallible(true)
+            .with_constraint(value::Kind::Map)
+    }
+}
+
+/// Parse the line as a separated list of key value pairs.
+fn parse_line<'a>(
+    input: &'a str,
+    key_value_delimiter: &'a str,
+    field_delimiter: &'a str,
+) -> IResult<&'a str, Vec<(String, Value)>> {
```

Comment:
nit: since we're only calling this function once, and we're ignoring the `&str` return value, we might as well ignore it in this function, so that the expression implementation is a bit cleaner.